expan/core/experiment.py

```diff
@@ -319,11 +319,12 @@ def chi_square_test_result_and_statistics(self, variant_column, weights, min_cou
 
         # Count number of observations per each variant
         variant_column = pd.Series(variant_column).dropna(axis=0)
-        variants_counts = variant_column.value_counts()
+        observed_freqs = variant_column.value_counts()
 
-        # Ensure at least a frequency of 5 at every location in observed_counts, otherwise drop the category
-        # see http://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.stats.chisquare.html
-        observed_freqs = variants_counts[variants_counts >= min_counts]
+        # Ensure at least a frequency of 5 at every location in observed_counts.
+        # It's recommended to not conduct test if frequencies in each category is less than 5
+        if len(observed_freqs[observed_freqs < min_counts]) >= 1:
+            raise ValueError("Chi-square test is not valid for small expected or observed frequencies.")
```

Comment:
@daryadedik @gbordyugov So what is a canonical way in Python to test input parameters?
asserts? or if-raise?

Reply:
as I see it for a second time already, I tag you both